flink-core/src/main/java/org/apache/flink/api/common/ExecutionConfig.java

```diff
@@ -612,6 +614,41 @@ public boolean hasGenericTypesDisabled() {
 		return disableGenericTypes;
 	}
 
+	/**
+	 * Enables the Flink runtime to auto-generated UID's for operators.
+	 *
+	 * @see #disableAutoGeneratedUIDs()
+	 */
+	public void enableAutoGeneratedUIDs() {
+		disableAutoGeneratedUid = false;
+	}
+
+	/**
+	 * Disable's auto-generated UID's forces users to manually specify UID's
+	 * on their datastreams using {@code DataStream#uid} or {@code DataStream#setUidHash}.
```

Comment:
Maybe let's not link `DataStream#setUidHash`? As this is last resort, a hackish approach. What do you think?